### PR TITLE
Feature/1.17 app show banner in homepage

### PR DESCRIPTION
**Pull request by wade-portkey**

### CODE REVIEW

File: packages/mobile-app-did/js/pages/DashBoard/Banner/index.tsx
Comment (Portkey-David) [inline packages/mobile-app-did/js/pages/DashBoard/Banner/index.tsx:1]: maybe this is better:  
 
  if(!list?.length) return null
  return <Carousel items={list} containerStyle={styles.containerer} />

Comment (Portkey-David) [reply packages/mobile-app-did/js/pages/DashBoard/Banner/index.tsx:1]: const list = homeBannerList.map(item => {
    return {
      url: item.url,
      imgUrl: getS3ImageUrl(item.imgUrl.filename_disk),
    };
  });
  
use useMemo to wrap list is better

Comment (wade-portkey) [reply packages/mobile-app-did/js/pages/DashBoard/Banner/index.tsx:1]: fixed

File: packages/mobile-app-did/js/pages/DashBoard/Header/index.tsx [hunk 1]
```diff
@@ -49,7 +86,16 @@ const DashBoardHeader: React.FC<DashBoardHeaderProps> = ({ title, scrollY }) =>
           titleStyle={{ fontSize: pTd(14) }}
         />
         {userInfo?.nickName ? (
-          <TextM style={styles.accountName}>{userInfo.nickName}</TextM>
+          <View style={styles.accountNameWrap}>
+            <TextM numberOfLines={1} style={[styles.accountName, { maxWidth: nickNameMaxWidth }]}>
```
Comment (Portkey-David) [inline packages/mobile-app-did/js/pages/DashBoard/Header/index.tsx:90]: Gstyle.minWidth()

File: packages/mobile-app-did/js/pages/Token/TokenDetail/index.tsx [hunk 2]
```diff
@@ -195,6 +200,13 @@ const TokenDetail: React.FC = () => {
     return balanceShow?.length > 18;
   }, [balanceShow]);
 
+  const bannerItemsList = getTokenDetailBannerList(tokenInfo.chainId, tokenInfo.symbol).map(item => {
+    return {
+      url: item.url,
+      imgUrl: getS3ImageUrl(item.imgUrl.filename_disk),
```
Comment (Portkey-David) [inline packages/mobile-app-did/js/pages/Token/TokenDetail/index.tsx:206]: usememo wrap ite

File: packages/mobile-app-did/js/pages/Token/TokenDetail/index.tsx [hunk 3]
```diff
@@ -245,6 +257,9 @@ const TokenDetail: React.FC = () => {
             />
           )}
         </View>
+        {bannerItemsList && bannerItemsList.length > 0 && (
```
Comment (Portkey-David) [inline packages/mobile-app-did/js/pages/Token/TokenDetail/index.tsx:260]: { bannerItemsList?.length > 0 && ... } is OK

Comment (wade-portkey) [reply packages/mobile-app-did/js/pages/Token/TokenDetail/index.tsx]: fixed

File: packages/mobile-app-did/js/pages/Token/TokenDetail/index.tsx (hunk 2 above)
Comment (wade-portkey) [reply packages/mobile-app-did/js/pages/Token/TokenDetail/index.tsx]: fixed

File: packages/mobile-app-did/js/pages/DashBoard/Header/index.tsx (hunk 1 above)
Comment (wade-portkey) [reply packages/mobile-app-did/js/pages/DashBoard/Header/index.tsx]: fixed

**Review by Portkey-David — APPROVED** [review]
ack

**Pull request merged**